.circleci/config.yml

```diff
@@ -500,39 +502,30 @@ jobs:
       - run:
           name: Set PYODIDE_BASE_URL
           command: |
+            PYODIDE_BASE_URL="https://cdn.jsdelivr.net/pyodide/v${CIRCLE_TAG}/pyc/" make dist/console.html
+            cp dist/console.html dist-pyc/console.html
             PYODIDE_BASE_URL="https://cdn.jsdelivr.net/pyodide/v${CIRCLE_TAG}/debug/" make dist/console.html
             cp dist/console.html dist-debug/console.html
             PYODIDE_BASE_URL="https://cdn.jsdelivr.net/pyodide/v${CIRCLE_TAG}/full/" make dist/console.html
       - run:
-          name: Deploy to pyodide-cdn2.iodide.io
+          name: Deploy to S3
           command: |
-            # Note: the following compression is the compression of files on S3
-            # to reduce storage size and outbound AWS traffic. The HTTP
-            # requests for these files made to S3 will result in a response
-            # with gzip compression. However because JsDelivr CDN proxies and caches these
-            # requests, this has no impact on the compression of zip files and
-            # wheels once the files are served to users via CDN.
-            find dist/ -type f -print0 | xargs -0 -n1 -I@ bash -c "echo \"Compressing @\"; gzip @; mv @.gz @;"
-            aws s3 sync dist/ "s3://pyodide-cdn2.iodide.io/v${CIRCLE_TAG}/full/" --exclude '*.zip' --exclude '*.whl' --exclude "*.tar" --cache-control 'max-age=30758400, immutable, public' --content-encoding 'gzip'    # 1 year cache
-            aws s3 sync dist/ "s3://pyodide-cdn2.iodide.io/v${CIRCLE_TAG}/full/" --exclude '*' --include '*.zip' --include "*.whl" --include "*.tar" --cache-control 'max-age=30758400, immutable, public'  --content-type 'application/wasm' --content-encoding 'gzip'  # 1 year
+            python3 tools/deploy_s3.py dist/ "v${CIRCLE_TAG}/full/" --bucket "pyodide-cdn2.iodide.io" --cache-control 'max-age=30758400, immutable, public'
```

Comment:
Good to see this deployment logic factored out and tested. In the future I think it would be good to move this whole recipe into a Python script that we can test (other than install requirements of course).